nimare/meta/cbmr.py

```diff
@@ -0,0 +1,1829 @@
+                self.inputs_["all_group_study_id"] = all_group_study_id
+                # collect studywise moderators if specficed
+                if self.moderators:
+                    if isinstance(self.moderators, str):
+                        self.moderators = [
+                            self.moderators
+                        ]  # convert moderators to a single-element list if it's a string
+                    all_group_moderators = dict()
+                    for group in all_group_study_id.keys():
+                        df_group = valid_dset_annotations.loc[
+                            valid_dset_annotations["study_id"].isin(all_group_study_id[group])
+                        ]
+                        group_moderators = np.stack(
+                            [df_group[moderator_name] for moderator_name in self.moderators],
+                            axis=1,
+                        )
+                        group_moderators = group_moderators.astype(np.float64)
+                        all_group_moderators[group] = group_moderators
+                    self.inputs_["all_group_moderators"] = all_group_moderators
+                # Calculate IJK matrix indices for target mask
+                # Mask space is assumed to be the same as the Dataset's space
+                # These indices are used directly by any KernelTransformer
+                all_foci_per_voxel, all_foci_per_study = dict(), dict()
+                for group in all_group_study_id.keys():
+                    group_study_id = all_group_study_id[group]
+                    group_coordinates = dataset.coordinates.loc[
+                        dataset.coordinates["study_id"].isin(group_study_id)
+                    ]
+                    # group-wise foci coordinates
+                    group_xyz = group_coordinates[["x", "y", "z"]].values
+                    group_ijk = mm2vox(group_xyz, mask_img.affine)
+                    group_foci_per_voxel = np.zeros(mask_img.shape, dtype=np.int32)
+                    for ijk in group_ijk:
+                        group_foci_per_voxel[ijk[0], ijk[1], ijk[2]] += 1
+                    # will not work with maskers that aren't NiftiMaskers
+                    group_foci_per_voxel = nib.Nifti1Image(
+                        group_foci_per_voxel, mask_img.affine, mask_img.header
+                    )
+                    group_foci_per_voxel = masker.transform(group_foci_per_voxel).transpose()
+                    # number of foci per voxel/study
+                    n_group_study = len(group_study_id)
+                    group_foci_per_study = np.array(
+                        [(group_coordinates["study_id"] == i).sum() for i in group_study_id]
+                    )
+                    group_foci_per_study = group_foci_per_study.reshape((n_group_study, 1))
+
+                    all_foci_per_voxel[group] = group_foci_per_voxel
+                    all_foci_per_study[group] = group_foci_per_study
```

Comment:
change to foci_per_voxel and foci_per_study

Reply:
Done :)